biorbd_optim/constraints.py

```diff
@@ -101,6 +108,25 @@ def __markers_to_pair(ocp, nlp, X, policy):
                 ocp.g_bounds.min.append(0)
                 ocp.g_bounds.max.append(0)
 
+    @staticmethod
+    def __align_with_custom_rt(ocp, nlp, X, policy):
+        """
+        Adds the constraint that the RT and the segment must be aligned at the desired instant(s).
+        :param nlp: An OptimalControlProgram class.
+        :param X: List of instant(s).
+        :param policy: Tuple of indices of segment and rt.
+        """
+        nq = nlp["dof_mapping"].nb_reduced
+        for x in X:
+            q = nlp["dof_mapping"].expand(x[:nq])
```

Comment:
Make sure that you use q_mapping 
Don't forget to add a test that will test this constraint